core/src/global/ionic-global.ts

```diff
@@ -1,16 +1,114 @@
-import { getMode, setMode, setPlatformHelpers } from '@stencil/core';
+import { getMode, setMode, setPlatformHelpers, getElement } from '@stencil/core';
+import { printIonWarning } from '@utils/logging';
 
-import type { IonicConfig, Mode } from '../interface';
+import type { IonicConfig, Mode, Theme } from '../interface';
 import { isPlatform, setupPlatforms } from '../utils/platform';
 
 import { config, configFromSession, configFromURL, saveConfig } from './config';
 
-// TODO(FW-2832): types
```

Comment:
This file still has several `any` types that need replacing, so I don't think this TODO should be removed yet.

Reply:
The remaining usages of `any` are valid and unlikely to be provided a more specific type. 

- `ref` - Stencil specific instance, there is no type for this. Stencil internally types the argument of `getMode`, which is `ref` as an `any`.
- `window` - we are applying a custom property to the window global. Casting as any is how we do this for Capacitor as well.
- `platformHelpers` - this type also is not exposed by Stencil and is passed to a Stencil utility function.